module/app/ante/bridge_fees.go

```diff
@@ -0,0 +1,150 @@
+package ante
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/keeper"
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/types"
+	sdkante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+)
+
+// BasisPointDivisor used in determining if a SendToEth fee meets the governance-controlled minimum
+const BasisPointDivisor int64 = 10000
+
+// BridgeFeeDecorator deducts fees from the first signer of the tx in the same way that cosmos-sdk's DeductFeeDecorator
+// will deduct fees, but only for MsgSendToEth. This is NOT meant to replace the DeductFeeDecorator!
+// If the first signer does not have the funds to pay for the fees, return with InsufficientFunds error
+// Call next AnteHandler if fees successfully deducted
+// CONTRACT: Tx must implement FeeTx interface to use BridgeFeeDecorator
+type BridgeFeeDecorator struct {
+	gravityKeeper  *keeper.Keeper
+	accountKeeper  *authkeeper.AccountKeeper
+	bankKeeper     *bankkeeper.BaseKeeper
+	feegrantKeeper *feegrantkeeper.Keeper // Note that gravity does not use the feegrant module, this should be nil
+}
+
+func NewBridgeFeeDecorator(gk *keeper.Keeper, ak *authkeeper.AccountKeeper, bk *bankkeeper.BaseKeeper, fk *feegrantkeeper.Keeper) BridgeFeeDecorator {
+	bfd := BridgeFeeDecorator{
+		gravityKeeper:  gk,
+		accountKeeper:  ak,
+		bankKeeper:     bk,
+		feegrantKeeper: fk, // Note that gravity does not use the feegrant module, this should be nil
+	}
+	bfd.ValidateMembers()
+
+	return bfd
+}
+
+func (bfd BridgeFeeDecorator) ValidateMembers() {
+	if bfd.gravityKeeper == nil {
+		panic("Nil gravityKeeper!")
+	}
+	if bfd.accountKeeper == nil {
+		panic("Nil accountKeeper!")
+	}
+	if bfd.bankKeeper == nil {
+		panic("Nil bankKeeper!")
+	}
+	// Gravity does not use the feegrant module, if it is added then this check should be enabled
+	// if bfd.feegrantKeeper == nil {
```

Comment:
maybe replace this with an assert that this value *is* nil. that way if anyone adds freegrantKeeper the code will panic/tests fail and they know to come here.